messages/teleporter/message_manager.go

```diff
@@ -125,29 +124,31 @@ func (m *messageManager) ShouldSendMessage(warpMessageInfo *vmtypes.WarpMessageI
 		return false, fmt.Errorf("relayer not configured to deliver to destination. destinationBlockchainID=%s", destinationBlockchainID.String())
 	}
 
+	teleporterMessenger := m.getTeleporterMessenger(destinationClient)
+	teleporterMessageID, err := m.calculateMessageID(teleporterMessenger, warpMessageInfo.WarpUnsignedMessage.SourceChainID, destinationBlockchainID, teleporterMessage.MessageNonce)
```

Comment:
It a bit sub-optimal that getting the message ID requires an additional API call here, but given the current interfaces and utilities, it's I agree that it's the easiest option. 

Alternatively, the message ID is emitted as a part of the `SendCrossChainMessage` event and could be used directly from there (though we don't currently subscribe or otherwise get that event), or we could provide a Golang utility for the hash operation, which would be far more efficient than an API call but have a small amount of duplicated logic.

I don't think we need to change it for this PR, but wanted to call out to hear people's thoughts.

Reply:
Yeah investigated this a bit and unfortunately there is no way to call any methods from the golang ABI without an RPC call, even if you're calling a `pure` function. I totally agree that needing the API here is very undesired.

I think the best solution here is probably to build our own Golang utility that reproduces the functionality. Importantly, we should add a test (or include the sanity check in the e2e network setup) that calls the Teleporter contract to make sure it aligns with our utility.

We can totally defer this to a new ticket, though.

Reply:
Also, the utility should be located in the `teleporter` repo.

Reply:
created new issue to track: https://github.com/ava-labs/awm-relayer/issues/136

Reply:
Agreed on all points. In the future when optimizing end to end relayer latency, minimizing the number of network round trips will be an obvious starting point.